Source/WTF/wtf/glib/Sandbox.cpp

```diff
@@ -36,10 +37,35 @@ bool isInsideFlatpak()
     return returnValue;
 }
 
-bool isInsideContainer()
+bool isInsideUnsupportedContainer()
 {
-    static bool returnValue = g_file_test("/run/.containerenv", G_FILE_TEST_EXISTS);
-    return returnValue;
+    static bool inContainer = g_file_test("/run/.containerenv", G_FILE_TEST_EXISTS);
+    static int supportedContainer = -1;
+
+    // Being in a container does not mean sub-containers cannot work. It depends upon various details such as
+    // docker vs podman, which permissions are given, is it privileged or unprivilged, and are unprivileged user namespaces enabled.
+    // So this just does a basic test of if `bwrap` runs successfully.
+    if (inContainer && supportedContainer == -1) {
+        const char* bwrapArgs[] = {
+            BWRAP_EXECUTABLE,
+            "--ro-bind", "/", "/",
+            "--proc", "/proc",
+            "--dev", "/dev",
+            "--unshare-all",
+            "true",
+            nullptr
+        };
+        GUniqueOutPtr<GError> error;
+        int waitStatus = 0;
+        gboolean spawnSucceeded = g_spawn_sync(nullptr, const_cast<char**>(bwrapArgs), nullptr,
+            G_SPAWN_STDERR_TO_DEV_NULL, nullptr, nullptr, nullptr, nullptr,
+            &waitStatus, &error.outPtr());
+        supportedContainer = spawnSucceeded && g_spawn_check_wait_status(waitStatus, &error.outPtr());
+        if (!supportedContainer)
+            g_warning("Bubblewrap does not work inside of this container, sandboxing will be disabled: %s", error->message);
```

Comment:
Too noisy. We shouldn't warn about this as it's expected and not an error. We should do at most a debug log here.

Reply:
I'd really like to keep the warning in some form that is disabling the sandbox. We are running webkit in podman on various distributions where it could work on one place and not another and a silent failure is counter productive.

Reply:
OK... try using WTFLogAlways then. It'll still be noisy, but at least somewhat less so than g_warning().